agent/accumulator.go

```diff
@@ -161,6 +164,14 @@ func (ac *accumulator) AddFields(
 	ac.metrics <- m
 }
 
+func (ac *accumulator) AddError(err error) {
+	if err == nil {
+		return
+	}
+	atomic.AddUint64(&ac.errCount, 1)
```

Comment:
this doesn't do what you think it does, I don't think there is a solution other than putting a mutex lock around this to avoid data races.